src/components/FilterList.js

```diff
@@ -1,53 +1,59 @@
 import './ShowList.css';
 import ShowList from './ShowList';
+import { useSelector } from "react-redux";
 
 
-function FilterList({ taskList, setTaskList , filter }){
+function FilterList(){
+
+const taskList = useSelector( state => state.taskList );
+const filter = useSelector( state =>  state.filter );
 
 const pending = filter.pending;
 const inProgress = filter.inProgress;
 const done = filter.done;
-    
+   
+
+
 
    
 
        return(
            <div>
             {
-                taskList.map((task) => {
+                taskList.map((task) => { 
```

Comment:
didn't like the idea of having these too many if and else checks can we make it more simpler ? 
Also I don't see any difference if `status` is whether in `pending`, `Inprogress` or `Done` state each time we are passsing same state of `status` as props, So why we need to check the status on first hand ?

Reply:
status is checked to filter the results while displaying the tasks and have tried other ways but if else statements only seem to work in this case

Reply:
I don't see any reason for having this much checks, since `ShowList` component only displays the content and update it on event. Try to use just pass the `task` state as it is and then let know where you are stuck.